tests/pipeline/test_engine.py

```diff
@@ -372,52 +381,59 @@ def test_masked_factor(self):
         values.
         """
         loader = self.loader
-        dates = self.dates[5:10]
+        dates = self.dates[5:8]
         assets = self.assets
         asset_ids = self.asset_ids
         constants = self.constants
         num_dates = len(dates)
         open = USEquityPricing.open
+        close = USEquityPricing.close
         engine = SimplePipelineEngine(
             lambda column: loader, self.dates, self.asset_finder,
         )
 
-        # These are the expected values for the OpenPrice factor. If we pass
-        # OpenPrice a mask, any assets that are filtered out should have all
-        # NaN values. Otherwise, we expect its computed values to be the
-        # asset's open price.
-        values = array([constants[open]] * num_dates, dtype=float)
-        missing_values = array([nan] * num_dates)
-
-        for asset_id in asset_ids:
-            mask = AssetID() <= asset_id
-            factor1 = OpenPrice(mask=mask)
-
-            # Test running our pipeline both with and without a second factor.
-            # We do not explicitly test the resulting values of the second
-            # factor; we just want to implicitly ensure that the addition of
-            # another factor to the pipeline term graph does not cause any
-            # unexpected exceptions when calling `run_pipeline`.
-            for factor2 in (None,
-                            RollingSumDifference(mask=NotSpecified),
-                            RollingSumDifference(mask=mask)):
-                if factor2 is None:
-                    columns = {'factor1': factor1}
-                else:
-                    columns = {'factor1': factor1, 'factor2': factor2}
-                pipeline = Pipeline(columns=columns)
-                results = engine.run_pipeline(pipeline, dates[0], dates[-1])
-                factor1_results = results['factor1'].unstack()
-
-                expected = {
-                    asset: values if asset.sid <= asset_id else missing_values
-                    for asset in assets
-                }
-
-                assert_frame_equal(
-                    factor1_results,
-                    DataFrame(expected, index=dates, columns=assets),
-                )
+        missing_values = full(num_dates, nan)
+        factor1_values = full(num_dates, constants[open], dtype=float64)
+        factor2_values = full(num_dates,
+                              3.0 * (constants[open] - constants[close]))
+
+        def create_expected_results(values):
```

Comment:
This would also make it easier to test with more complex masks.  See note below.